lib/vanagon/driver.rb

```diff
@@ -33,6 +35,19 @@ def initialize(platform, project, options = { :configdir => nil, :target => nil,
       raise Vanagon::Error.wrap(e, "Could not load the desired engine '#{engine}'")
     end
 
+    def filter_out_components(project, only_build)
+      filtered_components = []
+      only_build.each do |comp|
+        filtered_components += @project.filter_component(comp)
+      end
+      project.components = filtered_components.uniq
+      puts "Only building:" if @verbose
+      project.components.each do |comp|
+        puts comp.name if @verbose
+      end
```

Comment:
Tangent: we have GOT to decouple notification/logging from actual implementation. Vanagon needs a better `Logger`, stat.